src/components/table/daikin-table.ts

```diff
@@ -0,0 +1,408 @@
+   */
+  @state()
+  private _currentView: T[] = [];
+
+  private _updateCurrentView() {
+    const sort = this.sort;
+    if (!sort) {
+      this._currentView = Array.from(this.rows);
+      return;
+    }
+
+    const sortFunction = this.sortFunction?.[sort] ?? defaultSort;
+    this._currentView = this.rows.toSorted((a, b) => sortFunction(a, b, sort));
+
+    if (this.order === "desc") {
+      this._currentView.reverse();
+    }
+  }
+
+  private _emitChangeCheckEvent() {
+    this.dispatchEvent(new Event("change-check"));
+  }
+
+  private _handleHeaderCheckboxChange(): void {
+    switch (this._bulkRowsCheckState) {
+      case "checked":
+        this.selection = [];
+        break;
+
+      default:
+        this.selection = this._currentView.map(({ id }) => id);
+        break;
+    }
+
+    this._emitChangeCheckEvent();
+  }
+
+  private _handleBodyCheckboxChange(event: Event): void {
+    const { name: id } = event.target as DaikinCheckbox;
+
+    if (this.selection.includes(id)) {
+      this.selection = this.selection.filter((selectedId) => selectedId !== id);
+    } else {
+      this.selection = [...this.selection, id];
+    }
+
+    this._emitChangeCheckEvent();
+  }
+
+  private _handleClickSort(event: Event, key: keyof T): void {
+    event.stopPropagation();
+
+    if (this.sort === key) {
+      this.order = this.order === "asc" ? "desc" : "asc";
+    } else {
+      this.sort = key;
+      this.order = "asc";
+    }
+
+    this.dispatchEvent(new Event("change-sort"));
+  }
+
+  private _updateHeaderCheckboxState() {
+    const selectedIdLengthInPage = this.selection.length;
+
+    this._bulkRowsCheckState =
+      this._currentView.length === selectedIdLengthInPage
+        ? "checked"
+        : selectedIdLengthInPage
+          ? "indeterminate"
+          : "unchecked";
+  }
+
+  override render() {
+    const createHeaderRow = () =>
+      repeat(this.headers, ({ label, alignment, sortable, ...header }) => {
+        const key = String(header.key);
+
+        return html`<th
+          class="h-full p-0"
+          aria-sort=${ifDefined(
+            this.sortable && this.sort === key
+              ? this.order === "asc"
+                ? "ascending"
+                : "descending"
+              : undefined
+          )}
+        >
+          <slot
+            name=${`header:${key}`}
+            @change-sort=${(event: Event) =>
+              this._handleClickSort(event, header.key)}
+          >
+            <daikin-table-header-cell
+              alignment=${alignment ?? "left"}
+              ?sortable=${this.sortable && sortable}
+            >
+              ${label}
+            </daikin-table-header-cell>
+          </slot>
+        </th>`;
+      });
+
+    const createRow = (item: T) =>
+      repeat(
+        this.headers,
+        ({ alignment, ...header }) =>
+          html`<td class="h-full p-0">
+            <slot name=${`cell:${String(header.key)}:${item.id}`}>
```

Comment:
It is more natural to put the row before the column.

```suggestion
            <slot name=${`cell:${item.id}:${header.key}`}>
```